### PR TITLE
Add TO client ATS config funcs

### PR DESCRIPTION
**Pull request by rob05c**
#### What does this PR do?

Add TO client ATS config funcs

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [x] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] Other _________

#### Check all that apply

- [x] This PR includes tests
- [ ] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2724/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2727/
Test FAILed.

**Comment by dangogh** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2756/
Test FAILed.

**Comment by rob05c** [comment]
@dangogh Fixed

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2757/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2869/
Test FAILed.

File: traffic_ops/testing/api/v14/atsconfig_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	serverList, _, err := TOSession.GetServerByHostName(testServer.HostName)
+	if err != nil {
+		t.Fatalf("cannot GET Server: %v\n", err)
+	}
+	if len(serverList) < 1 {
+		t.Fatalf("cannot GET Server '" + testServer.HostName + "', returned no servers\n")
+	}
+	server := serverList[0]
+
+	// profiles, _, err := TOSession.GetProfileByName(server.Profile)
```
Comment (moltzaum) [inline traffic_ops/testing/api/v14/atsconfig_test.go]: Why are these commented out?

File: traffic_ops/client/atsconfig.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+func (to *Session) GetATSCDNConfig(cdnID int, fileName string) (string, ReqInf, error) {
+	return to.getConfigFile(apiBase + "/cdns/" + strconv.Itoa(cdnID) + "/configfiles/ats/" + fileName)
+}
+
+func (to *Session) GetATSCDNConfigByName(cdnName string, fileName string) (string, ReqInf, error) {
+	return to.getConfigFile(apiBase + "/cdns/" + cdnName + "/configfiles/ats/" + fileName)
+}
+
+func (to *Session) getConfigFile(uri string) (string, ReqInf, error) {
```
Comment (moltzaum) [inline traffic_ops/client/atsconfig.go]: Rename `GetConfigByURI`?

File: traffic_ops/client/atsconfig.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	defer resp.Body.Close()
+
+	data := tc.ATSConfigMetaData{}
+	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
+		return tc.ATSConfigMetaData{}, reqInf, err
+	}
+	return data, reqInf, nil
+}
+
+func (to *Session) GetATSServerConfigListByName(serverHostName string) (tc.ATSConfigMetaData, ReqInf, error) {
```
Comment (moltzaum) [inline traffic_ops/client/atsconfig.go]: This works but why use `to.request`?
```
func (to *Session) GetATSConfigMetadataByServer(server string) (*tc.ConfigFilesResponse, ReqInf, error) {
	data := tc.ATSConfigMetaData{}
	url := fmt.Sprintf("%s/servers/%s/configfiles/ats", apiBase, server)
	inf, err := get(to, url, &data)
	return &data, inf, err
}
```

**Comment by moltzaum** [comment]
I tested the PR by running the test `TestATSConfigs`. Every public-facing client function is run inside the test and they don't return any errors when I ran them. Not sure if trying to make them fail makes sense in this review.

Having this pulled in would be helpful for ATS config tests that depend on it. The client is common to all the config tests.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2909/
Test PASSed.

**Comment by moltzaum** [comment]
Where is this at? @rob05c do you plan on adding more? @dangogh in order to merge do you want to see anything else?

File: traffic_ops/testing/api/v14/atsconfig_test.go (hunk 1 above)
Comment (rob05c) [reply traffic_ops/testing/api/v14/atsconfig_test.go]: Fixed.

File: traffic_ops/client/atsconfig.go (hunk 2 above)
Comment (rob05c) [reply traffic_ops/client/atsconfig.go]: `getConfigFile` is an internal helper function. Funcs that begin with capitals are exported in Go. This shouldn't be exported.

**Comment by rob05c** [comment]
>Where is this at?
I think it's ready to be merged. I fixed your commented code comment, and commented on the capitalization comment. 

I don't object to using the `get` helper instead of `request`, but it'll take time to implement and verify. Personally I don't think it's big enough that it should prevent merging. But I'll try to find time, if this isn't merged shortly.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2944/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3217/
Test PASSed.

**Comment by ocket8888** [comment]
APIv1.4 no longer exists, and the API no longer serves configuration files at all - should this PR be closed?

**Pull request closed without merging**